### PR TITLE
Add wpa supplicant build tool

### PR DESCRIPTION
**Pull request by Pterosaur**
This PR depends on https://github.com/Azure/sonic-wpa-supplicant/pull/18, please wait for this PR merged.

Signed-off-by: Ze Gan <ganze718@gmail.com>

### CODE REVIEW

File: scripts/common/sonic-wpa-supplicant-build/build.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Install HIREDIS
+sudo apt-get install -y libhiredis0.14 libhiredis-dev
+
+# Install REDIS
+sudo apt-get install -y redis-server
+sudo sed -ri 's/^# unixsocket/unixsocket/' /etc/redis/redis.conf
+sudo sed -ri 's/^unixsocketperm .../unixsocketperm 777/' /etc/redis/redis.conf
+sudo sed -ri 's/redis-server.sock/redis.sock/' /etc/redis/redis.conf
+sudo service redis-server start
```
Comment (xumia) [inline scripts/common/sonic-wpa-supplicant-build/build.sh:11]: Can it be run in the docker container? One agent may run several test builds in parallel.

Comment (lguohan) [reply scripts/common/sonic-wpa-supplicant-build/build.sh:11]: agree, we should try to run this build in sonic-slave docker.

File: jenkins/common/sonic-wpa-supplicant-build-pr/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                dir('sonic-wpa-supplicant') {
+                    checkout([$class: 'GitSCM',
+                          branches: [[name: '${sha1}']],
+                          userRemoteConfigs: [[url: 'http://github.com/Azure/sonic-wpa-supplicant',
+                                               refspec: '+refs/pull/*:refs/remotes/origin/pr/*']]])
+                }
+                dir('sonic-mgmt-common') {
+                    checkout([$class: 'GitSCM',
+                          branches: [[name: 'refs/heads/master']],
+                          userRemoteConfigs: [[url: 'http://github.com/Azure/sonic-mgmt-common']]])
```
Comment (lguohan) [inline jenkins/common/sonic-wpa-supplicant-build-pr/Jenkinsfile:16]: i do not think sonic-mgmt-common is needed.

File: scripts/common/sonic-wpa-supplicant-build/build.sh (hunk 1 above)
Comment (lguohan) [reply scripts/common/sonic-wpa-supplicant-build/build.sh:11]: btw, @xumia, i think the build is running inside sonic-slave-buster.

Comment (xumia) [inline scripts/common/sonic-wpa-supplicant-build/build.sh:11]: The build is run inside the container "sonic-slave-buster", the installation should be good.

File: jenkins/common/sonic-wpa-supplicant-build-pr/Jenkinsfile (hunk 2 above)
Comment (Pterosaur) [reply jenkins/common/sonic-wpa-supplicant-build-pr/Jenkinsfile]: To be honest, I'm not sure.
According to my understanding, sonic-mgmt-common will be used to manager some configuration in the testing environment. So I included it.
But I will remove it according to your suggestion.

**Review by lguohan — APPROVED** [review]

File: jenkins/common/sonic-wpa-supplicant-build/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                          branches: [[name: 'refs/heads/master']],
+                          userRemoteConfigs: [[url: 'http://github.com/Azure/sonic-wpa-supplicant']]])
+                }
+                copyArtifacts(projectName: '../vs/buildimage-vs-all', filter: '**/*.deb', target: 'buildimage', flatten: false)
+            }
+        }
+
+        stage('Build') {
+            steps {
+                sh './scripts/common/sonic-wpa-supplicant/build.sh'
```
Comment (liushilongbuaa) [inline jenkins/common/sonic-wpa-supplicant-build/Jenkinsfile:30]: The PATH is different with the following script.

Comment (Pterosaur) [reply jenkins/common/sonic-wpa-supplicant-build/Jenkinsfile:30]: Fix it in this PR: https://github.com/Azure/sonic-build-tools/pull/195/
Please review it.

**Pull request merged**